pkg/ipcache/metadata.go

```diff
@@ -211,6 +220,14 @@ func (ipc *IPCache) InjectLabels(ctx context.Context, modifiedPrefixes []netip.P
 				// Already deleted, no new metadata to associate
 				continue
 			} // else continue below to remove the old entry
+
+			// Special case: if we are removing an IP from the reserved:host identity,
+			// we need to remove this prefix's labels from the local identity and
+			// force an update to the selector cache.
+			if oldID.ID == identity.ReservedIdentityHost {
+				i := ipc.updateReservedHostLabels(prefix, nil)
+				idsToAdd[i.ID] = i.Labels.LabelArray()
+			}
```

Comment:
I'll see if the blocks can be correctly joined without affecting logic.

Reply:
While thinking through this, I realized there was a potential bug when an IP lost reserved:node status. So the logic needed to be refactored to the "bottom" anyways.

Reply:
OK, check out the latest revision.

Reply:
Out of curiosity, how did you discover the potential bug?

Reply:
> Out of curiosity, how did you discover the potential bug?

I was thinking about more possible transitions (after our in-person conversation) and I realized I had missed a case. I also added this to the unit test.

As a hobby project, it would be fun to try and fuzz the ipcache in general. I'm thinking of a way to invent a sequence of operations and invariants.